File: src/main/java/com/sleepwalker/sleeplib/client/drawable/ModalFillSprite.java
package com.sleepwalker.sleeplib.client.drawable;

import com.sleepwalker.sleeplib.gg.essential.universal.UMatrixStack;
import org.jetbrains.annotations.NotNull;

import javax.annotation.Nonnull;
import java.awt.*;

public class ModalFillSprite extends ModalSprite {

    public ModalFillSprite(@NotNull TextureSource texture, int uOffset, int vOffset, int uWidth, int vHeight) {
        super(texture, uOffset, vOffset, uWidth, vHeight);
    }

    @Override
    public void drawImage(@Nonnull UMatrixStack ms, double x, double y, double width, double height, @Nonnull Color color) {
        texture.bind();
        innerBlit(ms.toMC(), color, (float) x, (float) (x + width), (float) y, (float) (y + height), 0, uWidth, vHeight, uOffset, vOffset, texture.getWidth(), texture.getHeight());
    }
}

File: src/main/kotlin/com/sleepwalker/sleeplib/elementa/drawable/ModalFillSprite.kt
package com.sleepwalker.sleeplib.elementa.drawable

import com.sleepwalker.sleeplib.gg.essential.universal.UMatrixStack
import java.awt.Color

class ModalFillSprite(texture: TextureSource, uOffset: Int, vOffset: Int, uWidth: Int, vHeight: Int) :
    ModalSprite(texture, uOffset, vOffset, uWidth, vHeight) {

    constructor(texture: TextureSource, uOffset: Int, vOffset: Int) : this(texture, uOffset, vOffset, texture.width, texture.height)
    constructor(texture: TextureSource) : this(texture, texture.width, texture.height)

    override fun drawImage(matrixStack: UMatrixStack, x: Double, y: Double, width: Double, height: Double, color: Color) {
        texture.bind()
        innerBlit(matrixStack, color, x, x + width, y, y + height, 0.0, uWidth, vHeight, uOffset.toDouble(), vOffset.toDouble(), texture.width, texture.height)
    }
}